src/data/dataset_interface.py

```diff
@@ -0,0 +1,28 @@
+import torchvision.transforms as transforms
+
+from abc import ABC, abstractmethod
+
+class TaskDataset(ABC):
+    @abstractmethod
+    def __init__(self,root_dir, kernel_size, sigma, size, batch_size=4, num_workers=2):
```

Comment:
Let's change the defaults to:
batch_size: 32
num_workers: 8

Feilong and I have the same amount of GPU VRAM and your CPU should be able to support this